From 984ca8246b67131f875e69c6726b6807a56aa3bf Mon Sep 17 00:00:00 2001
From: Jasper Orschulko <jasper@fancydomain.eu>
Date: Fri, 10 May 2024 12:41:35 +0200
Subject: [PATCH]  meta-secure-imx: Update layercompat

Signed-off-by: Jasper Orschulko <jasper@fancydomain.eu>
---
 conf/layer.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conf/layer.conf b/conf/layer.conf
index 0d0ca8c..cd0b9f2 100644
--- a/conf/layer.conf
+++ b/conf/layer.conf
@@ -12,7 +12,7 @@ LICENSE_PATH += "${LAYERDIR}/licenses"
 SECURE_BASE := '${LAYERDIR}'
 
 # Layer dependencies
-LAYERSERIES_COMPAT_secure-imx = "dunfell"
+LAYERSERIES_COMPAT_secure-imx = "dunfell kirkstone scarthgap"
 LAYERDEPENDS_secure-imx = "core openembedded-layer"
 
 # The dynamic-layers directory hosts the extensions and layer specific
-- 
2.45.0

