From 28613bafef058c30c74c0bb25a63bc93452fb3dd Mon Sep 17 00:00:00 2001
From: Stefano Babic <stefano.babic@swupdate.org>
Date: Sat, 25 May 2024 14:27:57 +0200
Subject: [PATCH 12/15] Update README

Signed-off-by: Stefano Babic <stefano.babic@swupdate.org>

Upstream-Status: Backported from Scarthgap to Mickledore
---
 README | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/README b/README
index 63cac53..ec803cb 100644
--- a/README
+++ b/README
@@ -12,6 +12,13 @@ This layer depends on:
 URI: git://github.com/openembedded/meta-openembedded.git
 subdirectory: meta-oe
 
+Documentation for the layer
+---------------------------
+
+Documentation for meta-swupdate is part of SWUpdate's docs.
+
+Check for http://sbabic.github.io/swupdate/building-with-yocto.html.
+
 BitBake variable expansion
 --------------------------
 
@@ -73,7 +80,7 @@ mechanism must exist to provide the passphrase.
 Maintainer
 ----------
 
-Stefano Babic <sbabic@denx.de>
+Stefano Babic <stefano.babic@swupdate.org>
 
 Submitting patches
 ------------------
-- 
2.45.1

