From 471f3ed4002ada5961052ea767f6d4ad2aceb9ba Mon Sep 17 00:00:00 2001
From: Lei Maohui <leimaohui@fujitsu.com>
Date: Mon, 13 Sep 2021 16:37:17 +0900
Subject: [PATCH 28/30] spdx-common.bbclass: Deleted redundant dependencies.

Signed-off-by: Lei Maohui <leimaohui@fujitsu.com>
---
 classes/spdx-common.bbclass | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/classes/spdx-common.bbclass b/classes/spdx-common.bbclass
index 2e195e6..abcd312 100644
--- a/classes/spdx-common.bbclass
+++ b/classes/spdx-common.bbclass
@@ -1,18 +1,5 @@
 # This class supplys common functions.
 
-
-SPDXEPENDENCY += "${PATCHTOOL}-native:do_populate_sysroot"
-SPDXEPENDENCY += " wget-native:do_populate_sysroot"
-SPDXEPENDENCY += " subversion-native:do_populate_sysroot"
-SPDXEPENDENCY += " git-native:do_populate_sysroot"
-SPDXEPENDENCY += " lz4-native:do_populate_sysroot"
-SPDXEPENDENCY += " lzip-native:do_populate_sysroot"
-SPDXEPENDENCY += " xz-native:do_populate_sysroot"
-SPDXEPENDENCY += " unzip-native:do_populate_sysroot"
-SPDXEPENDENCY += " xz-native:do_populate_sysroot"
-SPDXEPENDENCY += " quilt-native:do_populate_sysroot"
-SPDXEPENDENCY += " tar-native:do_populate_sysroot"
-
 SPDX_DEPLOY_DIR ??= "${DEPLOY_DIR}/spdx"
 SPDX_TOPDIR ?= "${WORKDIR}/spdx_sstate_dir"
 SPDX_OUTDIR ?= "${SPDX_TOPDIR}/${TARGET_SYS}/${PF}/"
@@ -393,5 +380,4 @@ python do_spdx_creat_tarball(){
     bb.note("Creat tarball for  " + spdx_outdir)
     tar_file = spdx_create_tarball(d, d.getVar('WORKDIR'), 'patched', spdx_outdir)
 }
-do_spdx[depends] = "${SPDXEPENDENCY}"
 
-- 
2.34.1

