From 57c570995307bb998b5ee8cd5d1d0e4b7cf13fce Mon Sep 17 00:00:00 2001
From: Lei Maohui <leimaohui@cn.fujitsu.com>
Date: Mon, 24 May 2021 17:34:18 +0900
Subject: [PATCH 14/30] Added delay after create_folder.

Signed-off-by: Lei Maohui <leimaohui@cn.fujitsu.com>
---
 classes/fossology-python.bbclass | 1 +
 1 file changed, 1 insertion(+)

diff --git a/classes/fossology-python.bbclass b/classes/fossology-python.bbclass
index d587920..bf1fd54 100644
--- a/classes/fossology-python.bbclass
+++ b/classes/fossology-python.bbclass
@@ -160,6 +160,7 @@ python do_upload(){
     if d.getVar('FOLDER_NAME', False):
         folder_name = d.getVar('FOLDER_NAME')
         folder = create_folder(d, foss, token, folder_name)
+        time.sleep(int(d.getVar('WAIT_TIME')))
         bb.note("folder = " + folder.name)
     else:
         folder = foss.rootFolder
-- 
2.34.1

