From efbb000d436c78b92db400444613ff224f971881 Mon Sep 17 00:00:00 2001
From: Lei Maohui <leimaohui@cn.fujitsu.com>
Date: Thu, 27 May 2021 12:28:38 +0900
Subject: [PATCH 15/30] Deleted repeated function.

Signed-off-by: Lei Maohui <leimaohui@cn.fujitsu.com>
---
 classes/fossology-python.bbclass | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/classes/fossology-python.bbclass b/classes/fossology-python.bbclass
index bf1fd54..2ba6e5b 100644
--- a/classes/fossology-python.bbclass
+++ b/classes/fossology-python.bbclass
@@ -241,20 +241,6 @@ def create_folder(d, foss, token, folder_name):
     else:
         return folder
 
-def get_upload(d, folder, foss):
-    filename = get_upload_file_name(d)
-    upload_list,pages = foss.list_uploads()
-    upload = None
-    bb.note("Check tarball, %s ,has been uploaded?" % filename)
-    for upload in upload_list:
-        bb.note("upload  = %s" % upload)
-        if upload.uploadname == filename and upload.foldername == folder.name:
-            bb.note("The size of uploaded file is %s" % upload.filesize)
-            bb.note("Found " + upload.uploadname  + " in " + folder.name)
-            bb.note("filesha1  = %s" % upload.filesha1)
-            return upload
-    return None
-
 python do_schedule_jobs(){
     import os
     import re
-- 
2.34.1

